mantidproject/mantid
# Engineering GUI GSAS tab add Refine All button

A button to run refinement on all runs loaded into the GSAS tab of the ED GUI has been added.

**To test:**

Some sample data is [here](https://github.com/mantidproject/mantid/files/1894002/GSASIIRefineFitPeaks.zip). You'll also need GSASII installed - go [here](http://docs.mantidproject.org/nightly/algorithms/GSASIIRefineFitPeaks-v1.html#installing-gsasii) for how to install GSASII.

Select the two Nexus files for **Focused Run Files** and click **Load**. Select the `.prm` file for **Instrument Parameter File**, the three `.cif`s for **Phase Files**, the location of `GSASIIscriptable.py` (part of GSASII) for **GSAS-II Installation Directory**, and choose somewhere to save out the GSAS-II project files for **New GSAS-II Project**. Fiddle with any other parameters (keep **RefinementMethod** as Rietveld), and click **Refine All**.

Make sure a refinement was completed for every run (you should see values in the **Fit Results** section and overplotted fit results in the **Preview** section. Try cancelling the algorithm and closing the GUI while refinement is running.

Fixes #22265 

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/commit/cd6a9ac0278e2e00ab5b389d4d421544880bc4c8)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
